VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -242,27 +254,31 @@ public void onClick(View v) {
 
         //support video download for video type
         final int totalCount = component.getBlockCount().videoCount;
-        if (totalCount == 0){
+        if (totalCount == 0 ){
             holder.noOfVideos.setVisibility(View.INVISIBLE);
             holder.bulkDownloadVideos.setVisibility(View.GONE);
         } else {
             holder.noOfVideos.setVisibility(View.VISIBLE);
             holder.noOfVideos.setText("" + totalCount);
 
-            int inProcessCount = dbStore.getVideosCountBySection(courseId, chapterId, sequentialId, null);
-            int webOnlyCount = dbStore.getWebOnlyVideosCountBySection(courseId, chapterId, sequentialId, null);
-            int videoCount = totalCount - inProcessCount - webOnlyCount;
-            if (videoCount > 0) {
-                holder.bulkDownloadVideos.setVisibility(View.VISIBLE);
-                holder.bulkDownloadVideos
-                    .setOnClickListener(new View.OnClickListener() {
-                        @Override
-                        public void onClick(View downloadView) {
-                            download(component.getVideos());
-                        }
-                    });
-            } else {
+            if (  row.numOfVideoNotDownloaded == 0 ){
                 holder.bulkDownloadVideos.setVisibility(View.GONE);
+            } else {
+                int inProcessCount = dbStore.getVideosCountBySection(courseId, chapterId, sequentialId, null);
+                int webOnlyCount = dbStore.getWebOnlyVideosCountBySection(courseId, chapterId, sequentialId, null);
```

Comment:
Out of curiosity, how fast is database access, will this stuff eventually want to be asynchronous?